parser_library/src/diagnostic.cpp

```diff
@@ -1477,6 +1474,14 @@ diagnostic_op diagnostic_op::warn_D025(const range& range, const std::string& ty
     return diagnostic_op(
         diagnostic_severity::warning, "D025", "The " + modifier + " modifier is ignored with type " + type, range);
 }
+diagnostic_op diagnostic_op::warn_D031(const range& range, const std::string& instr)
+{
+    return diagnostic_op(diagnostic_severity::warning,
+        "D031",
+        " Operand value must be relocatable symbol with instruction " + instr,
```

Comment:
I think it would be more useful to report the operand we're generating the warning for instead of the instruction name + replace must with should as this is a warning.